src/Microsoft.TestPlatform.CrossPlatEngine/Client/Parallel/ParallelProxyDiscoveryManager.cs

```diff
@@ -122,7 +120,7 @@ public bool HandlePartialDiscoveryComplete(IProxyDiscoveryManager proxyDiscovery
 
                 // Dispose concurrent executors
                 // Do not do the cleanuptask in the current thread as we will unncessarily add to discovery time
-                this.lastParallelDiscoveryCleanUpTask = Task.Run(() => this.UpdateParallelLevel(0));
```

Comment:
How does this change affect Discovery/Execution benchmarks? Should we validate with few scale test projects? cc: @navin22